porcupine/plugins/directory_tree.py

```diff
@@ -255,12 +262,13 @@ def refresh(self, junk: object = None) -> None:
 
     # The following two methods call each other recursively.
 
-    def _update_tags_and_content(self, project_root: Path, child_id: str) -> None:
+    def _update_tags_and_content(self, project_root: Path|None, child_id: str) -> None:
```

Comment:
It would be better to assert that it isn't `None` when calling this method. Every file belongs to some project and so the project path is never `None`.